bazel/README.md

```diff
@@ -256,8 +256,50 @@ The following optional features can be disabled on the Bazel build command-line:
 The following optional features can be enabled on the Bazel build command-line:
 
 * Exported symbols during linking with `--define exported_symbols=enabled`.
-  This is useful in cases where you have a lua script that loads shared object libraries, such as those installed via luarocks.
-* Perf annotation with `define perf_annotation=enabled` (see source/common/common/perf_annotation.h for details).
+  This is useful in cases where you have a lua script that loads shared object libraries, such as
+  those installed via luarocks.
+* Perf annotation with `--define perf_annotation=enabled` (see
+  source/common/common/perf_annotation.h for details).
+
+## Disabling extensions
+
+Envoy uses a modular build which allows extensions to be removed if they are not needed or desired.
+Extensions that can be removed are contained in
+[extensions_build_config.bzl](../source/extensions/extensions_build_config.bzl). Use the following
+procedure to customize the extensions for your build:
+
+* The Envoy build assumes that a Bazel repository named `@envoy_build_config` exists which
+  contains the file `@envoy_build_config//:extensions_build_config.bzl`. In the default build,
+  a synthetic repository is created containing [extensions_build_config.bzl](../source/extensions/extensions_build_config.bzl).
+  Thus, the default build has all extensions.
+* Start by creating a new Bazel workspace somewhere in the filesystem that your build can access.
+  This workspace should contain:
+  * Empty WORKSPACE file.
+  * Empty BUILD file.
+  * A copy of [extensions_build_config.bzl](../source/extensions/extensions_build_config.bzl).
+  * Comment out any extensions that you don't want to build in your file copy.
+
+To have your local build use your overridden configuration repository there are two options:
+
+1. Use the [`--override_repository`](https://docs.bazel.build/versions/master/command-line-reference.html)
+   CLI option to override the `@envoy_build_config` repo.
+2. Use the following snippet in your WORKSPACE before you load the Envoy repository. E.g.,
+
+```
+workspace(name = "envoy")
+
+local_repository(
+    name = "envoy_build_config",
+    path = "envoy-build-config",
+)
+
+local_repository(
+    name = "envoy",
+    path = "envoy",
```

Comment:
Does this assume some relative directory?

Reply:
I just pulled this from our Lyft stuff. Relative does work, but I will add some comments and make it more clear.